target/linux/ath79/dts/qca9558_belkin_f9j1108-v2.dts

```diff
@@ -0,0 +1,221 @@
+		label-mac-device = &eth1;
+	};
+	
+	leds {
+		compatible = "gpio-leds";
+		
+		usb {
+			label = "f9j1108-v2:blue:usb2";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&hub_port1>, <&hub_port0>;
+			linux,default-trigger = "usbport";
+		};
+		
+		wps-amber {
+			label = "f9j1108-v2:amber:wps";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+		
+		wps-blue {
+			label = "f9j1108-v2:blue:wps";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+		};
+
+		led_system: system {
+			label = "f9j1108-v2:blue:system";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+		};
+	};
+	
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		gpio_usb_power {
+			gpio-export,name = "f9j1108-v2:usb2:power";
+			gpio-export,output = <1>;
+			gpios = <&gpio 21 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+		
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&pcie1 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&usb_phy0 {
+	status = "okay";
+};
+
+&usb0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port0: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
+};
+
+&usb_phy1 {
+	status = "okay";
+};
+
+&usb1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port1: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		mtdparts: partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			
+			uboot: partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+			};
+			
+			partition@50000 {
+				compatible = "openwrt,okli";
+				label = "firmware";
+				reg = <0x050000 0xe20000>;
+			};
+			
+			partition@e70000 {
+				label = "loader";
+				reg = <0xe70000 0x180000>;
+			};
+			
+			art: partition@ff0000 {
+				label = "art";
+				reg = <0xff0000 0x010000>;
+			};
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+
+		qca,ar8327-initvals = <
+			0x04 0x00080080 /* PORT0 PAD MODE CTRL */
+			0x0c 0x07600000 /* PORT6 PAD MODE CTRL */
+			0x50 0xc737c737 /* LED_CTRL0 */
+			0x54 0x00000000 /* LED_CTRL1 */
+			0x58 0x00000000 /* LED_CTRL2 */
+			0x5c 0x0030c300 /* LED_CTRL3 */
+			0x7c 0x0000007e /* PORT0_STATUS */
+			0x94 0x0000007e /* PORT6 STATUS */
+			>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	mtd-mac-address = <&art 0x0>;
+	mtd-mac-address-increment = <(-1)>;
+	phy-handle = <&phy0>;
+	pll-data = <0x56000000 0x00000101 0x00001616>;
+
+	gmac-config {
```

Comment:
note to self: I don't understand what ``gmac-config`` does or why it should be used instead of ``fixed-link``

Reply:
If I understand correctly, it makes eth0 (in this case) associated with gmac1 instead of gmac0 (or maybe the reverse) and has parameters depending on the SOC for configuring the GMAC interfaces.

for ar934x there is parameters
switch-only-mode
switch-phy-swap

see page 153 in this datasheet
http://microchip.ho.ua/data/AR9344.pdf

Reply:
Nevermind the first part, which eth is associated with which gmac is done with `phy-handle`, but which PHY is connected to which gmac is decided by the gmac config.